src/Mod/Sketcher/Gui/DrawSketchDefaultHandler.h

```diff
@@ -0,0 +1,1003 @@
+    {
+        auto arc = std::make_unique<Part::GeomArcOfCircle>();
+        arc->setCenter(p1);
+        arc->setRange(start, end, true);
+        arc->setRadius(radius);
+        Sketcher::GeometryFacade::setConstruction(arc.get(), constructionMode);
+        ShapeGeometry.push_back(std::move(arc));
+    }
+
+    void addPointToShapeGeometry(Base::Vector3d p1, bool constructionMode)
+    {
+        auto point = std::make_unique<Part::GeomPoint>();
+        point->setPoint(p1);
+        Sketcher::GeometryFacade::setConstruction(point.get(), constructionMode);
+        ShapeGeometry.push_back(std::move(point));
+    }
+
+    void addEllipseToShapeGeometry(Base::Vector3d centerPoint,
+                                   Base::Vector3d majorAxisDirection,
+                                   double majorRadius,
+                                   double minorRadius,
+                                   bool constructionMode)
+    {
+        auto ellipse = std::make_unique<Part::GeomEllipse>();
+        ellipse->setMajorRadius(majorRadius);
+        ellipse->setMinorRadius(minorRadius);
+        ellipse->setMajorAxisDir(majorAxisDirection);
+        ellipse->setCenter(centerPoint);
+        Sketcher::GeometryFacade::setConstruction(ellipse.get(), constructionMode);
+        ShapeGeometry.push_back(std::move(ellipse));
+    }
+
+    void addCircleToShapeGeometry(Base::Vector3d centerPoint, double radius, bool constructionMode)
+    {
+        auto circle = std::make_unique<Part::GeomCircle>();
+        circle->setRadius(radius);
+        circle->setCenter(centerPoint);
+        Sketcher::GeometryFacade::setConstruction(circle.get(), constructionMode);
+        ShapeGeometry.push_back(std::move(circle));
+    }
+
+    void commandAddShapeGeometryAndConstraints()
+    {
+        auto shapeGeometry = toPointerVector(ShapeGeometry);
+        Gui::Command::doCommand(
+            Gui::Command::Doc,
+            Sketcher::PythonConverter::convert(Gui::Command::getObjectCmd(sketchgui->getObject()),
+                                               shapeGeometry)
+                .c_str());
+
+        auto shapeConstraints = toPointerVector(ShapeConstraints);
+        Gui::Command::doCommand(
+            Gui::Command::Doc,
+            Sketcher::PythonConverter::convert(Gui::Command::getObjectCmd(sketchgui->getObject()),
+                                               shapeConstraints)
+                .c_str());
+    }
+
+    void DrawShapeGeometry()
+    {
+        drawEdit(toPointerVector(ShapeGeometry));
+    }
+
+    void CreateAndDrawShapeGeometry()
+    {
+        createShape(true);
+        drawEdit(toPointerVector(ShapeGeometry));
+    }
+
+    //@}
+
+protected:
+    // The initial size may need to change in some tools due to the configuration of the tool, so
+    // resetting may lead to a different number than the compiled time value
+    int initialEditCurveSize;
+
+    std::vector<Base::Vector2d> EditCurve;

```

Comment:
@PaddleStroke 

Here I have a question for you.

At the very beginning, we mirrored the approach of the preexisting DrawSketchHandlers, in which often an EditCurve was provided, which would be drawn at each step based on Vector2D.

Later on, we introduced the ShapeGeometry and the DrawShapeGeometry, where essentially we add Part::Geometry to ShapeGeometry, and it gets Drawn  via DrawShapeGeometry. This way, avoiding to have to deal with the Vector2ds of EditCurve, and leveraging PythonConverter to generate all commands necessary at the end.

My question is: Could you take a look at all the DSHs we (mostly you) have written and evaluate whether we still need this "EditCurve" interface?

What I seek is:
If we do not need it, then we can drop it, including from the template parameters. We set for a more modern, less error-prone framework. What you see during construction is what you get, because it is actually based on that very same geometry.

If we do need it, then we keep it.

Please, let me know.

Reply:
For very basic tools, like line and circle, I seem to recall that it was still used. As those geometry are very simple. For example in the line tool instead of having a startPoint and endPoint, the editcurve was used to store the points.
But I think we can do without it. It will indeed probably be cleaner without it. We just make geometries and preview them.